compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirDeclarationCheckerUtils.kt

```diff
@@ -168,8 +175,30 @@ internal fun checkPropertyAccessors(
     reporter: DiagnosticReporter,
     context: CheckerContext
 ) {
-    property.setter?.source?.let {
-        if (property.isVal) {
+    if (property.isVal) {
+        if (property.getter != null) {
+            var reassignment: FirBackingFieldReference? = null
+            val visitor = object : FirVisitor<Unit, FirStatement?>() {
+                override fun visitElement(element: FirElement, data: FirStatement?) {
+                    element.acceptChildren(this, data)
```

Comment:
This checker is rejecting the following code while FE1.0 accepts it.

```
class Foo {
    val i: Int = 1
        get() {
            class Bar {
                var i: Int = 2
                    get() {
                        field++
                        return field
                    }
            }
            return field
        }
}
```

Reply:
Ah, so here we should check that backing field indeed points to the property of interest. Added more tests regarding local declarations to make sure it matches FE 1.0. Thanks!